src/middleware/auth.js

```diff
@@ -4,6 +4,9 @@ import jwt from 'jsonwebtoken'
 import User from '../domain/user.js'
 
 export async function validateTeacherRole(req, res, next) {
+  if (res.locals.skipTeacherValidation) {
```

Comment:
why skip teacherValidation?

Reply:
The only people allowed to modify existing user data is a teacher, or the student with their own user data.
This approach first checks if the logged in user has the same ID as the user data getting modified, if so, it is the user wanting to modify their own data, thus I choose to skip the teacher validation, since a user is allowed to modify their own data.

If I do not skip teacher validation, a student user cannot modify their own user data, since they are a student.

Let me know if you/Magnus/Nigel have better approaches for this problem.

Reply:
try to move all this validation into middleware.
either group or split the validation for teacher and ownership into one or seperate validations given owners will need to be able to update everything and teachers as well with the addition to role and cohort id. think also teachers has null for cohortid if im not mistaken. this is plainly a design choice, keep it middleware and handle them there accordingly